fs_attachment/tests/test_fs_storage.py

```diff
@@ -372,3 +373,42 @@ def test_recompute_urls(self):
         """
         )
         self.assertEqual(self.env.cr.dictfetchall()[0].get("count"), 2)
+
+    def test_url_for_image_dir_optimized_and_not_obfuscated(self):
+        # Create a base64 encoded mock image (1x1 pixel transparent PNG)
+        image_data = base64.b64encode(
+            b"\x89PNG\r\n\x1a\n\x00\x00\x00\rIHDR\x00\x00\x00\x01\x00\x00\x00\x01\x08"
+            b"\x06\x00\x00\x00\x1f\x15\xc4\x89\x00\x00\x00\nIDAT\x08\xd7c\xf8\x0f\x00"
+            b"\x01\x01\x01\x00\xd1\x8d\xcd\xbf\x00\x00\x00\x00IEND\xaeB`\x82"
+        )
+
+        # Create a mock image filestore
+        fs_storage = self.env["fs.storage"].create(
```

Comment:
What's the protocol of your storage? Protocol is a required field and there is nor more default value on it on the 16.0 branch. Be careful to don't serve your images from the odoo's filestore in a public way. If you configure your proxy to get access to the filestore under the 'images' path, you expose all the files from your odoo instance....

Reply:
@lmignon 

Thanks for clarifying, but I use different filestore for only products imsges thumbnails.

But the bug is completely different from these settings that you mentioned.

I can use many protocols and to test I used file protocol.

Thanks